sdk/typescript/introspector/test/invoke.spec.ts

```diff
@@ -51,31 +50,59 @@ describe("Invoke typescript function", function () {
     // Load function
     await load(files)
 
+    const scanResult = scan(files)
+
     // Mocking the fetch from the dagger API
     const input = {
-      objectName: "Bar",
-      methodName: "exec",
-      state: {},
-      inputs: {
+      parentName: "Bar",
+      fnName: "exec",
+      parentArgs: {},
+      fnArgs: {
         // string[]
         cmd: ["echo", "-n", "hello world"],
       },
     }
 
     // We wrap the execution into a Dagger connection
     await connection(async () => {
-      const result = await invoke(
-        input.objectName,
-        input.methodName,
-        input.state,
-        input.inputs
-      )
+      const result = await invoke(scanResult, input)
 
       // We verify the result, this could be serialized and set using `dag.ReturnValue` as a response
       assert.equal(result, "hello world")
     })
   })
 
+  it("Should correctly order arguments", async function () {
+    this.timeout(60000)
+
+    const files = await listFiles(`${rootDirectory}/multiArgs`)
+
+    // Load function
+    await load(files)
+
+    const scanResult = scan(files)
+
+    // Mocking the fetch from the dagger API
+    const input = {
+      parentName: "HelloWorld",
+      fnName: "compute",
+      parentArgs: {},
+      fnArgs: {
+        b: 2,
+        a: 4,
+        c: 3,
+      },
+    }
+
+    // We wrap the execution into a Dagger connection
+    await connection(async () => {
+      const result = await invoke(scanResult, input)
+
+      // We verify the result
+      assert.equal(result, 11)
+    })
```

Comment:
Is the connection really needed? If there's no core types, `.id()` won't be called on any. What's inside invoke that needs it then?

Reply:
That's right, but for consistency I preffered to keep the connection